modules/apps/notification/notification-test/src/testIntegration/java/com/liferay/notification/service/test/NotificationTemplateLocalServiceTest.java

```diff
@@ -51,25 +56,64 @@ public void setUp() throws Exception {
 	public void testAddNotificationTemplate() throws Exception {
 		User user = TestPropsValues.getUser();
 
-		try {
-			_notificationTemplateLocalService.addNotificationTemplate(
+		AssertUtils.assertFailure(
+			NotificationRecipientSettingNameException.class,
+			"Notification recipient setting name is invalid",
+			() -> _notificationTemplateLocalService.addNotificationTemplate(
 				NotificationTemplateUtil.createNotificationContext(
-					user, StringUtil.randomString(256),
-					NotificationConstants.TYPE_USER_NOTIFICATION));
-
-			Assert.fail();
-		}
-		catch (NotificationTemplateDescriptionException
-					notificationTemplateDescriptionException) {
-
-			Assert.assertEquals(
-				"The description cannot contain more than 255 characters",
-				notificationTemplateDescriptionException.getMessage());
-		}
+					TestPropsValues.getUser(), StringUtil.randomString(255),
+					RandomTestUtil.randomString(),
+					Arrays.asList(
+						NotificationTemplateUtil.
+							createNotificationRecipientSetting(
+								"from", "[%CURRENT_USER_EMAIL_ADDRESS%]"),
+						NotificationTemplateUtil.
+							createNotificationRecipientSetting(
+								"fromName", "[%CURRENT_USER_FIRST_NAME%]"),
+						NotificationTemplateUtil.
+							createNotificationRecipientSetting(
+								"to", "test@liferay.com"),
+						NotificationTemplateUtil.
+							createNotificationRecipientSetting(
+								"userScreenName", "userScreenName")),
+					StringUtil.randomString(255),
+					NotificationConstants.TYPE_EMAIL)));
+		AssertUtils.assertFailure(
+			NotificationRecipientSettingNameException.class,
+			"Notification recipient setting name is invalid",
+			() -> _notificationTemplateLocalService.addNotificationTemplate(
+				NotificationTemplateUtil.createNotificationContext(
+					user, RandomTestUtil.randomString(),
+					StringUtil.randomString(255),
+					Collections.singletonList(
+						NotificationTemplateUtil.
+							createNotificationRecipientSetting(
+								"singleRecipient", user.getScreenName())),
+					RandomTestUtil.randomString(),
+					NotificationConstants.TYPE_USER_NOTIFICATION)));
+
+		AssertUtils.assertFailure(
+			NotificationTemplateDescriptionException.class,
+			"The description cannot contain more than 255 characters",
+			() -> _notificationTemplateLocalService.addNotificationTemplate(
+				NotificationTemplateUtil.createNotificationContext(
+					user, RandomTestUtil.randomString(),
+					StringUtil.randomString(256),
```

Comment:
Please use the `NotificationTemplateUtil.createNotificationContext(User user, String description, String type) {`